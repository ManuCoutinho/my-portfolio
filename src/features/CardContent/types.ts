import { DataProps } from 'features/Card/types'

export type MenuLinkProps = {
  color1: string
  color2: string
}

export type CardContentProps = DataProps
