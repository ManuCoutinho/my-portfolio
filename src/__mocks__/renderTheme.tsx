import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
