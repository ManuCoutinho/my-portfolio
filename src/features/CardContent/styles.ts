import styled, { css } from 'styled-components'
import * as Tabs from '@radix-ui/react-tabs'

export const Header = styled.header`
  ${({ theme }) => css`
    width: 100%;
    border-bottom: 1px solid ${theme.colors.info};
    margin-bottom: 0.8rem;
    transition: ${theme.transition};
    h4 {
      color: ${theme.colors.text};
      text-transform: uppercase;
      padding: 1rem;
      margin-left: 1rem;
    }
  `}
`

export const TabTitle = styled(Tabs.Trigger)`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    margin-bottom: 0.5rem;
    outline: 0;
    border: 0;
    background: none;
    font-size: 1.25rem;
    transition: ${theme.transition};
    &[data-state='active'] {
      color: ${theme.colors.active};
    }
    &:hover {
      transform: scale(1.03);
      transition: ${theme.transition};
    }
  `}
`
export const Content = styled(Tabs.Content)`
  ${() => css`
    outline: 0;
    overflow: hidden;
  `}
`
