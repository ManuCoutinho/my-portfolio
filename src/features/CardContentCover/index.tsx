import { StackBox } from 'components/StackBox'
import dynamic from 'next/dynamic'

import { CardContentCoverProps } from './types'

const Slider = dynamic(() => import('components/Slider'))

const CardContentCover: React.FC<CardContentCoverProps> = ({ img }) => {
  return (
    <StackBox id='overview' direction='column'>
      <Slider img={img} />
    </StackBox>
  )
}

export default CardContentCover
