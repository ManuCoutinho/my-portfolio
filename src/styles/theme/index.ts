import { DefaultTheme, CustomTheme } from 'styled-components'
import dark from './dark'
import light from './light'

export type Theme = typeof defaultTheme

const defaultTheme = {
	font: {
		cursive:
			"'Montserrat Alternates', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif",
		regular:
			"'Josefin Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif"
	},
	fontSize: {
		xsm: '0.5rem', //8px
		sm: '0.875rem', //14px
		md: '1rem', //16px
		wd: '1.25rem', //20px
		lg: '1.5rem', //24px
		xl: '2rem', //36px
		xlg: '3rem' //48px
	},
	fontWeight: {
		thin: 100,
		light: 200,
		regular: 400,
		semibold: 600,
		bold: 700
	},
	radius: {
		sm: '8px',
		md: '16px',
		pill: '25px',
		full: '50%'
	}
} as const

function combineTheme(theme: CustomTheme): DefaultTheme {
	return { ...defaultTheme, ...theme }
}
export { combineTheme, dark, light }
