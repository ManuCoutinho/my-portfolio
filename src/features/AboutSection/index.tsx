import Image from 'next/image'
import { Description } from './Description'
import { Carousel } from './CarouselAbout'
import { StackBoxCenter, Title } from 'components/Layout/Base'
import { AboutSection, ContainerAbout } from './styles'
import { useTheme } from 'styled-components'

import lightImg from '../../../public/assets/perfil.webp'
import darkImg from '../../../public/assets/perfil-green.webp'

const About: React.FC = () => {
  const theme = useTheme()

  return (
    <AboutSection id='about'>
      <Title>Sobre</Title>
      <ContainerAbout>
        <StackBoxCenter>
          <Image
            width='300'
            height='300'
            src={theme.title === 'light' ? lightImg : darkImg}
            loading='lazy'
            alt='girl coding with cup of coffee'
          />
        </StackBoxCenter>
        <Description />
      </ContainerAbout>
      <Carousel />
    </AboutSection>
  )
}

export default About
