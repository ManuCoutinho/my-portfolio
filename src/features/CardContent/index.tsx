import dynamic from 'next/dynamic'
import * as Tabs from '@radix-ui/react-tabs'
import { Heading } from 'components/Heading'
import { StackBox } from 'components/StackBox'
import * as Styled from './styles'
import { CardContentProps } from './types'

const Overview = dynamic(() => import('features/CardContentCover'))
const Details = dynamic(() => import('features/CardContentDetails'))

export const CardContent: React.FC<CardContentProps> = ({ ...props }) => {
  return (
    <Tabs.Root defaultValue='overview'>
      <Styled.Header>
        <Heading as='h4' size='small'>
          {props.name}
        </Heading>
        <Tabs.List>
          <StackBox justify='space-evenly' gap={2}>
            <Styled.TabTitle value='overview'>Vis√£o geral</Styled.TabTitle>
            <Styled.TabTitle value='details'>Detalhes</Styled.TabTitle>
          </StackBox>
        </Tabs.List>
      </Styled.Header>
      <Styled.Content value='overview'>
        <Overview img={props.img} />
      </Styled.Content>
      <Styled.Content value='details'>
        <Details {...props} />
      </Styled.Content>
    </Tabs.Root>
  )
}
