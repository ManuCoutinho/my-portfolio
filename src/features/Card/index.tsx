import { useEffect, useState, FC } from 'react'
import { useRouter } from 'next/router'
import dynamic from 'next/dynamic'
import * as Dialog from '@radix-ui/react-dialog'
import { useTheme } from 'styled-components'
import { StackBox } from 'components/StackBox'
import { Heading } from 'components/Heading'
import { DataProps } from './types'
import * as Styled from './styles'
import { CardContent } from 'features/CardContent'

const Modal = dynamic(() => import('components/Dialog'), { ssr: false })

export const Card: FC<DataProps> = ({ ...props }) => {
  const router = useRouter()
  const { title } = useTheme()
  const [modalIsOpen, setModalIsOpen] = useState(false)
  const [isTouchDevice, setIsTouchDevice] = useState(false)

  const handleCloseModal = () => {
    router.push('/#portfolio')
  }

  useEffect(() => {
    if (typeof window !== 'undefined' && window.innerWidth <= 830) {
      setIsTouchDevice(true)
    }
  }, [router])
  function handleOpenDetails() {
    isTouchDevice
      ? router.push({
          pathname: `${props.id}`,
          query: { id: props.id }
        })
      : setModalIsOpen(true)
  }
  return (
    <StackBox>
      <Dialog.Root open={modalIsOpen} onOpenChange={setModalIsOpen}>
        <Styled.CardComponent type='button' onClick={handleOpenDetails}>
          <Styled.Image>
            <picture>
              <img
                loading='lazy'
                width={300}
                height={350}
                alt={props.alt}
                src={
                  title === 'light' ? props.coverAltLight : props.coverAltDark
                }
              />
              <source
                srcSet={
                  title === 'light'
                    ? props.coverDefaultLight
                    : props.coverDefaultDark
                }
                type='image/webp'
              />
            </picture>
          </Styled.Image>
          <Heading size='small' as='h3'>
            {props.name}
          </Heading>
          <Dialog.Trigger style={{ border: 0 }}>
            <Modal onClose={handleCloseModal}>
              <CardContent {...props} />
            </Modal>
          </Dialog.Trigger>
        </Styled.CardComponent>
      </Dialog.Root>
    </StackBox>
  )
}
