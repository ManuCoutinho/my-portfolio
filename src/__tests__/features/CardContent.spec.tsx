import { screen } from '@testing-library/react'
import { render } from '../../styles/theme/renderTheme'
import mockData from '../../__mocks__/data'
import { CardContent } from '../../features/CardContent'

describe('<CardContent />', () => {
  it('should render CardContent section components correctly', () => {
    render(<CardContent {...mockData} />)
    expect(
      screen.getByRole('heading', { name: mockData.name })
    ).toBeInTheDocument()
  })
  it('should render tabs components', () => {
    render(<CardContent {...mockData} />)
    expect(
      screen.getByRole('tab', { name: /visÃ£o geral/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('tab', { name: /detalhes/i })).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /img1/i })).toBeInTheDocument()
    expect(screen.getByRole('tabpanel')).toBeInTheDocument()
  })
  it('should match to snapshot', () => {
    const { container } = render(<CardContent {...mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
