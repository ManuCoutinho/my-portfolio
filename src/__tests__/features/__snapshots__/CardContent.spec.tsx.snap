// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContent /> should match to snapshot 1`] = `null`;
